sdk/src/utils.ts

```diff
@@ -598,8 +601,60 @@ export const calculateFeeTierWithLinearRatio = (tickCount: bigint): FeeTier => {
   return newFeeTier(tickCount * toPercentage(1n, 4n), tickCount)
 }
 
-export const assert = (condition: boolean, message?: string) =>  {
+export const assert = (condition: boolean, message?: string) => {
   if (!condition) {
     throw new Error(message || 'assertion failed')
   }
-}
\ No newline at end of file
+}
+
+export const calculateConcentration = (tickSpacing: number, minimumRange: number, n: number) => {
+  const concentration = 1 / (1 - Math.pow(1.0001, (-tickSpacing * (minimumRange + 2 * n)) / 4))
+  return concentration / CONCENTRATION_FACTOR
+}
+
+export const calculateTickDelta = (
+  tickSpacing: number,
+  minimumRange: number,
+  concentration: number
+) => {
+  const base = Math.pow(1.0001, -(tickSpacing / 4))
+  const logArg =
+    (1 - 1 / (concentration * CONCENTRATION_FACTOR)) /
+    Math.pow(1.0001, (-tickSpacing * minimumRange) / 4)
+
+  return Math.ceil(Math.log(logArg) / Math.log(base) / 2)
+}
+
+export const getConcentrationArray = (
```

Comment:
Export already calculated arrays for all `FEE_TIERS`. it is always static, there is no point in counting it on the frontend